pytorch_lightning/trainer/trainer.py

```diff
@@ -634,6 +643,18 @@ def run_evaluation(self, test_mode: bool = False, max_batches=None):
 
         return eval_loop_results, deprecated_eval_results
 
+    def track_output_for_epoch_end(self, outputs, output):
```

Comment:
can we maybe not rely on call by reference but instead make the function return the current output on the correct device and detached? If we then append it like it was before this would be easier to follow because you can skip looking into the function if you don't care for these modifications.

Reply:
Done !